src/script/server/api/services/screenshot/service.js

```diff
@@ -0,0 +1,86 @@
+const chromeLauncher = require('chrome-launcher')
+const chromeRemoteInterface = require('chrome-remote-interface')
+const isEmpty = require('lodash/isEmpty')
+const log = require('loglevel')
+
+const reducers = require('../../../../redux/reducers')
+
+const launchChrome = async (url) => {
+  const chrome = await chromeLauncher.launch({
+    startingUrl: url,
+    chromeFlags: ['--disable-gpu', '--headless'],
+  })
+  return chrome
+}
+
+const connectToChrome = async (port) => {
+  // check chrome version
+  const versionInfo = await chromeRemoteInterface.Version({ port })
+  log.info(`chrome version found : ${versionInfo.Browser}`)
+  const versionNumber = versionInfo.Browser.match(/\d+/g)[0] // HeadlessChrome/62.0.3185.0
+  if (versionNumber < 62) {
+    throw Error(
+      'Your Chrome version is too old, you need at least version 62 to capture screenshots with react-workbench.'
+    )
+  }
+
+  // connect to chrome
+  const client = await chromeRemoteInterface({ port })
+  const { DOM, Page } = client
+  await Page.enable()
+  await DOM.enable()
+  return client
+}
+
+const getComponentBox = async (client) => {
+  const { DOM } = client
+  const { root } = await DOM.getDocument()
+  const { nodeId } = await DOM.querySelector({
+    selector: '#component > *',
+    nodeId: root.nodeId,
+  })
+  const { model } = await DOM.getBoxModel({ nodeId })
+  return {
+    x: model.content[0],
+    y: model.content[1],
+    width: model.width,
+    height: model.height,
+    scale: 1,
+  }
+}
+
+const captureScreenshot = async (client, metrics) => {
+  const { Page, Emulation } = client
+  // set screen / device metrics
+  if (!isEmpty(metrics)) await Emulation.setDeviceMetricsOverride(metrics)
+  // get component bounding box
+  const componentBox = await getComponentBox(client)
+  // capture component screenshot
+  const { data } = await Page.captureScreenshot({ clip: componentBox })
+  return Buffer.from(data, 'base64')
+}
+
+/**
+ * Capture a screenshot of the tested component
+ * @param {*} metrics see https://chromedevtools.github.io/devtools-protocol/tot/Emulation/#method-setDeviceMetricsOverride
+ */
+const capture = metrics => async (dispatch, getState) => {
+  const { PORT } = reducers.config.get()(getState())
+  const url = `http://localhost:${PORT}`
+  let chrome
+  let client
+  try {
+    log.info(`launch chrome headless on ${url}`)
+    chrome = await launchChrome(url)
```

Comment:
I thought about it, I did it in that way because if an error occured or at the end, I need to close the client and kill the chrome instance.
If I do it in the compose way, I have to return the client and chrome instance, and always try...catch to close the process if an error occured.